Production_scripts/Costs-02-district_nursing.R

```diff
@@ -0,0 +1,368 @@
+
+
+
+## save outfile ##
+# .zsav
+#haven::write_sav(dn_raw_costs_contacts,
+#                 paste0(get_slf_dir(), "/Costs/raw_dn_costs_with_contacts.sav"),
+#                 compress = TRUE
+#)
+
+
+
+
+# ----------------------------------------------------------
+
+
+## Calculate population cost for NHS Highland with HSCP population ratio. ##
+# Of the two HSCPs, Argyll and Bute provides the District Nursing data which is 27% of the population.
+
+lookup <- haven::read_sav(paste0(get_lookups_dir(), "/Populations/Estimates/HSCP2019_pop_est_1981_2020.sav"))
+
+
+# Select only the HSCPs for NHS Highland & years since 2015
+lookup <-
+  lookup %>%
+  filter(HSCP2019 == "S37000004" | HSCP2019 == "S37000016") %>%
+  filter(Year > 2015 | Year == 2015)
+
+
+# Create year as FY = YYYY from CCYY.
+lookup <-
+  lookup %>%
+  rename(calendar_year = Year) %>%
+  mutate(year = convert_year_to_fyyear(calendar_year))
+
+
+## outfile ##
+outfile <-
+  lookup %>%
+  select(year, HSCP2019, Pop) %>%
+  group_by(year, HSCP2019) %>%
+  summarise(pop = sum(Pop))
+
+
+
+# HSCP name
+outfile <-
+  outfile %>%
+  mutate(HSCPName = hscp_to_hscpnames(HSCP2019))
+
+
+# add Health Board code
+outfile <-
+  outfile %>%
+  mutate(HB2019 = "S08000022")
+
+
+outfile <-
+  outfile %>%
+  group_by(year, HB2019) %>%
+  mutate(total_pop = sum(pop))
+
+
+## compute proportion ##
+outfile <-
+  outfile %>%
+  mutate(pop_proportion = pop / total_pop) %>%
+  mutate(pop_pct = pop_proportion * 100)
+
+
+## Argyll and Bute is the only HSCP in NHS Highland that submits data ##
+outfile <-
+  outfile %>%
+  filter(HSCPName == "Argyll and Bute")
+
+
+
+## match files ##
+
+# run if not currently loaded from above #
+# read raw costs with contacts
+# dn_raw_costs_contacts <- haven::read_sav(paste0(get_slf_dir(), "/Costs/raw_dn_costs_with_contacts.sav"))
+
+
+# match
+data <-
+  outfile %>%
+  full_join(dn_raw_costs_contacts, by = c("HB2019", "year"))
+
+
+
+# recode NA pop_proportion with 1
+data <-
+  data %>%
+  mutate(pop_proportion = replace_na(pop_proportion, 1))
+
+
+
+## total net cost ##
+data <-
+  data %>%
+  mutate(cost_total_net = ((cost * 1000) / (number_of_contacts / pop_proportion))
+         ) %>%
+  # sort by HB2019
+  arrange(HB2019) %>%
+  # keep only records with cost
+  filter(!is.na(cost_total_net))
+
+
+
+# ----------------------------------------------------------
+
+
+## Fix incomplete submisssions ##
+# If a Partnership has abnormally low contacts this will affect the cost so use the previous year until we have a complete submission
+
+# explore the trends
+
+data <-
+  data %>%
+  group_by(Board_Name) %>%
+  mutate(max_contacts = max(number_of_contacts)) %>%
+  mutate(pct_of_max = number_of_contacts / max_contacts * 100)
+
+
+
+# plot #
+ggplot(data = data, aes(x = year, y = pct_of_max, group = Board_Name)) +
+  geom_line(aes(color = Board_Name)) +
+  labs(color = "NHS Board", x = "Year")
+
+
+
+## upflift any 'copied' costs ##
+data <-
```

Comment:
```suggestion
matched_costs<- matched_data
```
This could be renamed here to matched_costs